compiler/rustc_parse/src/parser/expr.rs

```diff
@@ -595,6 +606,27 @@ impl<'a> Parser<'a> {
                 let operand_expr = this.parse_dot_or_call_expr(Default::default())?;
                 this.recover_from_prefix_increment(operand_expr, pre_span, starts_stmt)
             }
+            // Recover from `--x`:
+            token::BinOp(token::Minus)
+                if this.look_ahead(1, |t| *t == token::BinOp(token::Minus))
+                    && !this.token.can_begin_expr() =>
```

Comment:
This is a part of a `match` on `this.token.uninterpolate().kind` ⇒ `this.token` is always `token::BinOp(token::Minus)` ⇒ `!this.token.can_begin_expr()` is always `false` ⇒ this is a dead branch.

Reply:
oh I did not see that